src/preprocessor.ts

```diff
@@ -1,13 +1,29 @@
 import { ParserOptions, parse as babelParser } from '@babel/parser';
 import traverse, { NodePath } from '@babel/traverse';
 import { ImportDeclaration, isTSModuleDeclaration } from '@babel/types';
-
+import { parse } from '@vue/compiler-sfc';
 import { PrettierOptions } from './types';
 import { getCodeFromAst } from './utils/get-code-from-ast';
 import { getExperimentalParserPlugins } from './utils/get-experimental-parser-plugins';
 import { getSortedNodes } from './utils/get-sorted-nodes';
 
 export function preprocessor(code: string, options: PrettierOptions) {
+    if (options.filepath?.endsWith('.vue')) return code
+    return process(code, options)
+}
+
+export function vuePreprocessor(code: string, options: PrettierOptions) {
+    const { descriptor } = parse(code);
+    const content = (descriptor.script ?? descriptor.scriptSetup)?.content ?? code
+
+    return code.replace(
+        content,
+        `\n${process(content, options)}\n`,
+    );
+
+}
+
+function process(code: string, options: PrettierOptions) { {
```

Comment:
I would suggest to make a preprocessor folder and separate them in different files. I would say `process` also can go in a separate file. Maybe `preprocessor` can also be renamed to something like `defaultPreprocessor`? What do you think?

Reply:
Yeah i've done the changes as requested. I'm happy with whatever you feel is best. Your repository after all :)